os-server/src/main/java/app/openschool/coursemanagement/api/mapper/CourseOfUserMapper.java

```diff
@@ -0,0 +1,64 @@
+package app.openschool.coursemanagement.api.mapper;
+
+import app.openschool.coursemanagement.api.dto.CourseOfUserDto;
+import app.openschool.coursemanagement.entity.Course;
+import app.openschool.coursemanagement.entity.Module;
+import app.openschool.coursemanagement.entity.ModuleItem;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+public class CourseOfUserMapper {
+
+  public static List<CourseOfUserDto> toCourseOfUserDtoList(List<Course> courseList) {
+    List<CourseOfUserDto> courseOfUserDtoList = new ArrayList<>();
+    for (Course course : courseList) {
+      courseOfUserDtoList.add(toCourseOfUserDto(course));
+    }
+    return courseOfUserDtoList;
+  }
+
+  public static CourseOfUserDto toCourseOfUserDto(Course course) {
+
+    String courseStatus = course.getStatus().getType();
+    String buttonName = courseStatus.equals("COMPLETED") ? "RATE COURSE" : "RESUME COURSE";
+
+    if (courseStatus.equals("COMPLETED")) {
+      return new CourseOfUserDto(course.getTitle(), courseStatus, 100, buttonName);
+    }
+
+    Optional<Long> remainingTimeOfCourse =
+        course.getModules().stream()
+            .filter(module -> module.getStatus().getType().equals("IN_PROGRESS"))
+            .map(Module::getModuleItems)
+            .flatMap(Collection::stream)
+            .filter(moduleItem -> moduleItem.getStatus().getType().equals("IN_PROGRESS"))
+            .map(ModuleItem::getEstimatedTimeImMinutes)
+            .reduce(Long::sum);
+
+    String remainingTime =
```

Comment:
not a good idea to store in the string, it is better to use time-based types or long milliseconds, the FE will convert it to the required data type.

Reply:
Since the user will estimate time of item I think we shouldn't suggest to estimate item in milliseconds. I think it will be better that the estimated time will be in long minutes. What do you think about this?

Reply:
nope, they will interact with usual date formats but we will story them as milliseconds